x/concentrated-liquidity/client/query_proto_wrap.go

```diff
@@ -0,0 +1,167 @@
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	userPositions, err := q.Keeper.GetUserPositions(ctx, sdkAddr, req.PoolId)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	positions := make([]model.PositionWithUnderlyingAssetBreakdown, 0, len(userPositions))
+
+	for _, position := range userPositions {
+		// get the pool from the position
+		pool, err := q.Keeper.GetPoolFromPoolIdAndConvertToConcentrated(ctx, position.PoolId)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+
+		asset0, asset1, err := cl.CalculateUnderlyingAssetsFromPosition(ctx, position, pool)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+
+		// Append the position and underlying assets to the positions slice
+		positions = append(positions, model.PositionWithUnderlyingAssetBreakdown{
+			Position: position,
+			Asset0:   asset0,
+			Asset1:   asset1,
+		})
+	}
+
+	return &clquery.UserPositionsResponse{
+		Positions: positions,
+	}, nil
+}
+
+// PositionById returns a position with the specified id.
+func (q Querier) PositionById(ctx sdk.Context, req clquery.PositionByIdRequest) (*clquery.PositionByIdResponse, error) {
+	position, err := q.Keeper.GetPosition(ctx, req.PositionId)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	positionPool, err := q.Keeper.GetPoolFromPoolIdAndConvertToConcentrated(ctx, position.PoolId)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	asset0, asset1, err := cl.CalculateUnderlyingAssetsFromPosition(ctx, position, positionPool)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &clquery.PositionByIdResponse{
+		Position: model.PositionWithUnderlyingAssetBreakdown{
+			Position: position,
+			Asset0:   asset0,
+			Asset1:   asset1,
+		},
+	}, nil
+}
+
+// Pools returns all concentrated pools in existence.
+func (q Querier) Pools(
+	ctx sdk.Context,
+	req clquery.PoolsRequest,
+) (*clquery.PoolsResponse, error) {
+	anys, pageRes, err := q.Keeper.GetSerializedPools(ctx, req.Pagination)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &clquery.PoolsResponse{
+		Pools:      anys,
+		Pagination: pageRes,
+	}, nil
+}
+
+// Params returns module params
+func (q Querier) Params(ctx sdk.Context, req clquery.ParamsRequest) (*clquery.ParamsResponse, error) {
+	return &clquery.ParamsResponse{Params: q.Keeper.GetParams(ctx)}, nil
+}
+
+// TotalLiquidityForRange returns an array of LiquidityDepthWithRange, which contains the range(lower tick and upper tick) and the liquidity amount in the range.
+func (q Querier) TotalLiquidityForRange(ctx sdk.Context, req clquery.TotalLiquidityForRangeRequest) (*clquery.TotalLiquidityForRangeResponse, error) {
+	liquidity, err := q.Keeper.GetTickLiquidityForFullRange(
+		ctx,
+		req.PoolId,
+	)
```

Comment:
The description implies the user specifies a lower and upper tick to get the liquidity range for, but in the function itself we are calling tick liquidity for the full range only (I know this was just copied over, but still unsure of which logic this should be utilizing)

Reply:
I see the proto only expects a pool ID, but I'm confused on if the future intention is to allow users to provide a tick range?

Reply:
Agree this doesn't make much sense. However, Jacky is integrating right now so it's best to check in with him first separately to avoid breaking him.

Made an issue to track: https://github.com/osmosis-labs/osmosis/issues/4958

Reply:
Renamed: [7fada3b](https://github.com/osmosis-labs/osmosis/pull/4956/commits/7fada3b1adfeb58ee2c014b9859069e22db93fa0)